### ISSUE TITLE
Scatters are rendered outside of grid when setting axis range

### ISSUE THREAD

**Issue by NingMengMeng** [apache/incubator-echarts#10010]
### Version
4.2.1-rc1

### Steps to reproduce
When `min` and `max` are set for `yAxis`, those outside the range of `min` and `max` are rendered, which is not expected.

```js
option = {
    xAxis: {},
    yAxis: {
        min: 5,
        max: 10
    },
    series: [{
        symbolSize: 20,
        data: [
            [10.0, 8.04],
            [8.0, 6.95],
            [13.0, 7.58],
            [9.0, 8.81],
            [11.0, 8.33],
            [14.0, 9.96],
            [6.0, 7.24],
            [4.0, 4.26],
            [12.0, 10.84],
            [7.0, 4.82],
            [5.0, 5.68]
        ],
        type: 'scatter'
    }]
};
```
### What is expected?
希望低于min和高于max的不要渲染出来
Scatters are not rendered for those outside grid.

### What is actually happening?
They are rendered outside grid.

**Comment by Ovilia** [comment]
除了散点图之外，所有图表类型都应该做这个检查。工作量较大，下个版本再支持。

**Comment by pissang** [comment]
Added `clip` option in the commit https://github.com/apache/incubator-echarts/commit/0b2b5e65b9d7b56a8bcac80a4d795aef8d097fa7 .It  will be released in the 4.4.0
